test/e2e/network/endpointslice.go

```diff
@@ -303,6 +304,131 @@ var _ = SIGDescribe("EndpointSlice", func() {
 		deleteEndpointSlices(cs, f.Namespace.Name, svc2)
 		expectEndpointsAndSlices(cs, f.Namespace.Name, svc2, []*v1.Pod{pod1, pod2}, 2, 2, true)
 	})
+
+	ginkgo.It("should NOT include terminating endpoints for Endpoints and EndpointSlices", func() {
+		labelPod1 := "pod1"
+		labelValue := "on"
+
+		pod1 := podClient.Create(&v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "pod1",
+				Labels: map[string]string{
+					labelPod1: labelValue,
+				},
+			},
+			Spec: v1.PodSpec{
+				// use long grace periods to leave time to test endpoints
+				TerminationGracePeriodSeconds: utilpointer.Int64Ptr(120),
+				Containers: []v1.Container{
+					{
+						Name:    "container1",
+						Image:   imageutils.GetE2EImage(imageutils.BusyBox),
+						Command: []string{"sleep", "1000000"},
+						Ports: []v1.ContainerPort{{
+							Name:          "example-name",
+							ContainerPort: int32(3000),
+						}},
+					},
+				},
+			},
+		})
+
+		svc1 := createServiceReportErr(cs, f.Namespace.Name, &v1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "example-int-port",
+			},
+			Spec: v1.ServiceSpec{
+				Selector: map[string]string{labelPod1: labelValue},
+				Ports: []v1.ServicePort{{
+					Name:       "example",
+					Port:       80,
+					TargetPort: intstr.FromInt(3000),
+					Protocol:   v1.ProtocolTCP,
+				}},
+			},
+		})
+
+		// wait for pod to start up
+		err := wait.Poll(5*time.Second, time.Minute, func() (bool, error) {
+			var err error
+			pod1, err = podClient.Get(context.TODO(), pod1.Name, metav1.GetOptions{})
+			if err != nil {
+				return false, err
```

Comment:
do we want to fail or to retry if the  Get fails?

Reply:
I think Get failing is fine here, since we know the previous Create call succeeded already.